plaso/parsers/mactime.py

```diff
@@ -50,176 +56,257 @@ def __init__(self):
 
-    # Check if the following columns contain a base 10 integer value if set.
-    for column_name in ('uid', 'gid', 'size'):
-      column_value = row.get(column_name, None)
-      if not column_value:
-        continue
+    line_number = 0
 
-      try:
-        int(column_value, 10)
-      except (TypeError, ValueError):
-        return False
+    try:
+      line = line_reader.readline()
+    except UnicodeDecodeError as exception:
+      raise errors.UnableToParseFile(
+          'unable to read line: {0:d} with error: {1!s}'.format(
+              line_number, exception))
+
+    while line:
+      values = line.split('|')
+      if len(values) < 11:
+        error_string = 'invalid number of values in line: {0:d}'.format(
+            line_number)
+        if line_number == 0:
+          raise errors.UnableToParseFile(error_string)
+
+        parser_mediator.ProduceExtractionWarning(error_string)
 
-    for column_name in ('atime', 'mtime', 'ctime', 'crtime'):
-      column_value = row.get(column_name, None)
-      if not column_value:
-        continue
+      else:
+        md5_value = values.pop(0)
+        if md5_value == '0':
+          md5_value = None
+        elif not self._MD5_RE.match(md5_value):
+          error_string = 'invalid MD5 value in line: {0:d}'.format(line_number)
+          if line_number == 0:
+            raise errors.UnableToParseFile(error_string)
+
+          parser_mediator.ProduceRecoveryWarning(error_string)
+
+        crtime_value = self._GetLastValueAsFloatingPoint(
+            parser_mediator, values, 'creation time', line_number)
+        ctime_value = self._GetLastValueAsFloatingPoint(
+            parser_mediator, values, 'inode change time', line_number)
+        mtime_value = self._GetLastValueAsFloatingPoint(
+            parser_mediator, values, 'modification time', line_number)
+        atime_value = self._GetLastValueAsFloatingPoint(
+            parser_mediator, values, 'access time', line_number)
+
+        size_value = self._GetLastValueAsBase10Integer(
+            parser_mediator, values, 'size', line_number)
+        gid_value = self._GetLastValueAsBase10Integer(
+            parser_mediator, values, 'group identifier (GID)', line_number)
+        uid_value = self._GetLastValueAsBase10Integer(
+            parser_mediator, values, 'user identifier (UID)', line_number)
+
+        if uid_value is not None:
+          # Note that the user_sid value is expected to be a string or None.
+          uid_value = '{0:d}'.format(uid_value)
+
+        mode_as_string_value = values.pop(-1) or None
+
+        inode_value = values.pop(-1) or None
+        if '-' in inode_value:
+          inode_value, _, _ = inode_value.partition('-')
+
+        try:
+          inode_value = int(inode_value, 10)
+        except (TypeError, ValueError):
+          inode_value = None
+          parser_mediator.ProduceRecoveryWarning(
+              'invalid inode value in line: {0:d}'.format(line_number))
```

Comment:
Please include inode_value in the warning message

Reply:
done